libraries/lib-cloud-audiocom/OAuthService.h

```diff
@@ -0,0 +1,113 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  OAuthService.h
+
+  Dmitry Vedenko
+
+**********************************************************************/
+#pragma once
+
+#include <chrono>
+#include <functional>
+#include <string_view>
+#include <mutex>
+
+#include "Observer.h"
+
+namespace cloud::audiocom
+{
+class ServiceConfig;
+
+//! Message that is sent when authorization state changes. This message can be
+//! sent from any thread.
```

Comment:
That comment above makes me nervous.  `Observer.h` says `intended for single-threaded use only`.  Should you use `CallAfter` wherever you publish, and change the comment to say, "Messages are received in the main thread in idle time" ?

Reply:
No, I'd rather stop using Observer.

Reply:
Or, should there be mutual exclusion around all `Publish` calls?  What might stil happen is that a Subscription is destroyed concurrently with a publication.  That might not have bad consequences, but I would have to reexamine Observer.h to be sure of that.

Reply:
I have reviewed the usages, it can be moved to main thread for now.